[workloadmeta] Extract the org.opencontainers.image.revision docker label

jybp opened the pull request:
### What does this PR do?

Automatically extract the `org.opencontainers.image.revision` docker label into the `git.commit.sha` tag.

### Motivation

The `git.commit.sha` tag must be set for Datadog's Source Code Integration setup.
Our public documentation currently suggest using the `org.opencontainers.image.revision` docker label and to configure the agent to extract the tag. See [Configure the Agent](https://docs.datadoghq.com/integrations/guide/source-code-integration/?tab=dockerruntime#configure-the-agent).  
 This PR will make the Source Code Integration setup easier by allowing to skip the `Configure the Agent` part.

### Additional Notes

The label is added inside the `standardDockerLabels` map since we do want this behavior to be enabled by default for all.

I've opened a draft [documentation PR](https://github.com/DataDog/documentation/pull/12237/files).

### Describe how to test/QA your changes

Start the Agent with a traced app running within a container with the label `org.opencontainers.image.revision` set.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno)
  has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] If applicable, docs team has been notified or
  [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).

jybp committed:
[workloadmeta] Extract the org.opencontainers.image.revision docker label into the git.commit.sha tag

juliogreff commented:
@jybp the tagger currently only has tags with underscores instead of dots. Any reason why we couldn't use `git_commit_sha` as the tag instead of `git.commit.sha`?

juliogreff commented on pkg/tagger/collectors/workloadmeta_extract.go:
```diff
@@ -78,9 +80,10 @@ var (
+		dockerLabelEnv:      tagKeyEnv,
+		dockerLabelVersion:  tagKeyVersion,
+		dockerLabelService:  tagKeyService,
+		dockerLabelRevision: tagKeyGitCommitSha,
```
Standard tags in this context mean [Unified Service Tagging](https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging/?tab=kubernetes). I'm not sure if the commit ID applies here, and would recommend moving it to `lowCardOrchestratorLabels` maybe.

ahmed-mez replied:
definitely +1

ahmed-mez replied:
Let's also rename it `containerRevisionLabel` as it can be found on container runtimes other than Docker [based on this](https://github.com/opencontainers/image-spec/blob/main/annotations.md)

lraucy replied:
Thanks for catching this! I updated the PR to move it in the lowCard list. I also got rid of the constants to be consistent with the rest of the map, let me know if you want me to re-introduce constants for this.

ahmed-mez replied:
no I think keeping it in the low card map makes sense, thanks!

lraucy commented:
> @jybp the tagger currently only has tags with underscores instead of dots. Are we sure we couldn't use `git_commit_sha` as the tag instead of `git.commit.sha`?

What is your concern?
The reason why we're going with `git.commit.sha` is because it's already existing as an "attribute" in other Datadog products (in CI Visibility spans), so we preferred to go with a common name; esp. because they both represent the same data.

lraucy committed:
[workloadmeta] Move git.commit.sha in lowCard tags.

Standard tags are for Unified Service Tagging, and git.commit.sha has
nothing to do with that; so lowCard tags is more appropriate.

ahmed-mez commented on releasenotes/notes/opencontainers-image-revision-as-tag-caa73d54488f0b52.yaml:
```diff
@@ -0,0 +1,4 @@
+---
+enhancements:
+  - |
+    Automatically extract the org.opencontainers.image.revision docker label into the git.commit.sha tag.
```
```suggestion
    Automatically extract the ``org.opencontainers.image.revision`` container label into the ``git.commit.sha`` tag.
```

lraucy committed:
Update releasenotes/notes/opencontainers-image-revision-as-tag-caa73d54488f0b52.yaml

Co-authored-by: Ahmed Mezghani <38987709+ahmed-mez@users.noreply.github.com>
